targets/targets.json

```diff
@@ -251,7 +251,7 @@
         "extra_labels": ["NXP", "LPC176X", "MBED_LPC1768", "NXP_EMAC"],
         "supported_toolchains": ["ARM", "uARM", "GCC_ARM", "IAR"],
         "detect_code": ["1010"],
-        "device_has": ["RTC", "USTICKER", "ANALOGIN", "ANALOGOUT", "CAN", "DEBUG_AWARENESS", "EMAC", "ETHERNET", "I2C", "I2CSLAVE", "INTERRUPTIN", "LOCALFILESYSTEM", "PORTIN", "PORTINOUT", "PORTOUT", "PWMOUT", "SEMIHOST", "SERIAL", "SERIAL_FC", "SLEEP", "SPI", "SPISLAVE", "STDIO_MESSAGES", "FLASH"],
```

Comment:
Is there any reasonable way to have "MPU" on by default for the cores that have it? If it's easy for devices to opt out, how many devices would want to be opting out?

The possible analogue here is SYSTICK, where a couple of platforms define `NO_SYSTICK`.

Reply:
The initial implementation had this on by default. With the switch to a formal HAL specification this was changed so that everything with an ARMv7-M and ARMv8-M MPU must add ```MPU``` to ```device_has```. The goal with this is to remain consistent with the other HAL specifications and list and enable supported features with ```device_has```.

> If it's easy for devices to opt out, how many devices would want to be opting out?

I would imagine not many would want to opt out. Right now you can't opt in or opt out. Everything with an ARMv7-M or ARMv8-M MPU must add ```MPU``` to ```device_has``` or they get a ```#error```. I did this to prevent new targets from accidentally forgetting to turn this setting on.

Reply:
So why is it a target setting at all, in the current version? If it's that strongly on, I'd be inclined to follow `SYSTICK`. MPU is a core feature, for mbed OS core control. Opt outs exceptional, and if we ever support the opt-out, it's `NO_MPU`.

Reply:
Okay, that makes sense now with the `MPU_CUSTOM`. Without that, there was no way to actually disable the standard v7/v8 ones. Complete replacement by "forking" the standard ones as custom will probably be a more maintainable solution in the short term than target and core code trying not to step on each other's toes while both messing with the MPU.